File: Source/Fantasy.Jobs/Management/InProcessJobEngineHost.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Jobs.Management
{
    public class InProcessJobEngineHost : MarshalByRefObject
    {
        public override object InitializeLifetimeService()
        {
            return null;
        }



        public void Run(JobManager manager, Guid jobId)
        {

            (new JobManagerAccessor()).SetJobManager(manager);

            JobEngine engine = new JobEngine(jobId);
            engine.Initialize();
            engine.WaitForExit();
        }
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/management/InProcessJobEngineHost.java
package fantasy.jobs.management;

import java.util.UUID;

import fantasy.jobs.JobEngine;

public class InProcessJobEngineHost
{
	
	public final void Run(JobManager manager, UUID jobId) throws Exception
	{

		(new JobManagerAccessor()).SetJobManager(manager);

		JobEngine engine = new JobEngine(jobId);
		engine.Initialize();
		engine.WaitForExit();
	}
}